packages/algob/test/internal/core/config/config-loading.ts

```diff
@@ -169,4 +190,79 @@ describe("config loading", function () {
       ).catch((err) => console.log(err)); ;
     });
   });
+
+  describe("load kmd accounts", function () {
+    useFixtureProject("config-project");
+    useEnvironment();
+
+    const fakeKmd: Kmd = {} as Kmd; // eslint-disable-line @typescript-eslint/consistent-type-assertions
+    const kmdOp = new KMDOperatorMock(fakeKmd);
+    let net: NetworkConfig;
+
+    beforeEach(function () {
+      net = this.env.config.networks.kmdNet;
+    });
+
+    afterEach(function () {
+      kmdOp.resetAccounts();
+    });
+
+    it("should ignore if kmd config is not defined", async function () {
+      net.kmdCfg = undefined;
+      const result = await _loadKMDAccounts(net, kmdOp);
+      assert.isUndefined(result);
+    });
+
+    it("should not connect to kmd if config is invalid", async function () {
+      const invalidKmdCfg = net.kmdCfg as KmdCfg;
+      invalidKmdCfg.port = 123; // invalid port
+
+      expectBuilderErrorAsync(
+        () => _loadKMDAccounts(net, new KMDOperator(createKmdClient(invalidKmdCfg))),
+        ERRORS.KMD.CONNECTION
+      ).catch((err) => console.log(err)); ;
+    });
+
+    it("network accounts should take precedence over KMD accounts", async function () {
+      const networkAcc = net.accounts[0];
+      const commonName = networkAcc.name;
+
+      // insert kmd account with same name but different address
+      const kmdAcc = {
+        name: commonName,
+        addr: 'some-different-addr',
+        sk: new Uint8Array(kmdOp.skArray)
+      };
+      kmdOp.addKmdAccount(kmdAcc);
+      await _loadKMDAccounts(net, kmdOp);
+
+      assert.notInclude(net.accounts, kmdAcc); // network.accounts should not include kmd account with same name
+      assert.include(net.accounts, networkAcc); // network.accounts should include networkAcc (higher precendence)
+    });
+
+    it("kmd accounts with different names should be successfully merged", async function () {
+      kmdOp.addKmdAccount({ name: "some-different-name-1", addr: "some-addr-1", sk: new Uint8Array(kmdOp.skArray) });
+      kmdOp.addKmdAccount({ name: "some-different-name-2", addr: "some-addr-2", sk: new Uint8Array(kmdOp.skArray) });
+
+      await _loadKMDAccounts(net, kmdOp);
+
+      for (const a of net.accounts) {
+        assert.include(net.accounts, a); // assert if network accounts are merged
+      }
```

Comment:
`an array will always have it's own element in it`
this will always be true. right ?

Reply:
Yes, that's correct. Just checked here for sanity purpose. Looked more clear to check for all accounts. Let's wait for @robert-zaremba to see this.

Reply:
did you mean sanity testing ? https://www.edureka.co/blog/what-is-sanity-testing/
or i have misunderstood

Reply:
Yes, since `_loadKMDAccounts` is merging kmd accounts into network.accounts, and some accounts are ignored based on name conflict, i wanted to do a check for all accounts. But yes, i know network.accounts have higher precendence and won't be ignored. Maybe we can remove this.

Reply:
yes, But In these lines:
consider an example
```
for (const element of array) {
        assert.include(array, element);
}
```
No matter what operations are done previously, this will always be true as `we are picking element from array and checking if it is in the array`

Reply:
Ok, let's remove